android/jni/com/mapswithme/maps/Framework.cpp

```diff
@@ -826,12 +826,14 @@ Java_com_mapswithme_maps_Framework_nativeShowCountry(JNIEnv * env, jclass, jstri
 JNIEXPORT void JNICALL
 Java_com_mapswithme_maps_Framework_nativeSetRoutingListener(JNIEnv * env, jclass, jobject listener)
 {
+  CHECK(g_framework, ("Framework isn't created yet!"));
```

Comment:
Есть, почему, если будут ещё какие-то листенеры добавляться и похожий креш случится - выдаст осмысленную ошибку в лог

> On May 11, 2016, at 1:49 PM, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/Framework.cpp <https://github.com/mapsme/omim/pull/3167#discussion_r62827264>:
> 
> > @@ -826,12 +826,14 @@ Java_com_mapswithme_maps_Framework_nativeShowCountry(JNIEnv * env, jclass, jstri
> >  JNIEXPORT void JNICALL
> >  Java_com_mapswithme_maps_Framework_nativeSetRoutingListener(JNIEnv * env, jclass, jobject listener)
> >  {
> > +  CHECK(g_framework, ("Framework isn't created yet!"));
> В этих проверках же нет смысла, не так ли?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/3167/files/4554bd2e3a81b5e98d268979d0c3fecf24618824#r62827264>